person.rb

```diff
@@ -0,0 +1,28 @@
+require './nameable'
+
+class Person < Nameable
+  attr_accessor :name, :age, :parent_permission
+  attr_reader :id
+
+  def initialize(age, name = 'unknown', parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def generate_id
+    rand(1..1000)
+  end
```

Comment:
- [ ]  As previous reviewer suggested You should make sure that the person class contain a method called `correct_name `that return the name attribute.